python/kserve/test/test_server.py

```diff
@@ -381,7 +382,14 @@ def test_list_handler(self, http_server_client):
     def test_health_handler(self, http_server_client):
         resp = http_server_client.get('/v1/models/TestModel')
         assert resp.status_code == 200
-        assert resp.content == b'{"name":"TestModel","ready":"True"}'
+        # for some reason the RayServer responds with the stringified python bool
+        # when run on pydantic < 2 and the bool when run on pydantic >= 2
+        # eg {"name":"TestModel","ready":"True"} vs {"name":"TestModel","ready":true}
```

Comment:
I think it is a different bug altogether. Before moving to fastapi, it seems the [health handler returned bool values](https://github.com/kserve/kserve/blob/21ce62bf4522e63ff6a60435250c5cbd55b12b43/python/kserve/kserve/handlers/health.py#L25-L47), 
but after the migration it started [returning string values because how the type is defined](https://github.com/kserve/kserve/blob/e6eede2ea0e94cbe75e90785877cba33f9014216/python/kserve/kserve/protocol/rest/v1_endpoints.py#L39) ?
@yuzisun

Reply:
The code seems to be returning bool, as @timothyjlaurent mentioned the issue is for the rayserve case?

Reply:
This is for the RayServe case.

In the previous assertion with pydantic v1 - we were getting `{"name":"TestModel","ready":"True"}` the string value, and testing for that. Seems like it was a bug that got "blessed" by enshrinining it in a testcase.
With the pyanticv2 we're (correctly) getting a bool value for the `ready` key (not a stringified python string) - This, I believe is the correct behavior we should expect.